### ISSUE TITLE
Create Trestle pipelines

### ISSUE THREAD

**Issue by smlambert** [adoptium/ci-jenkins-pipelines#895]
We are trialing a way for JVM developers from WG member companies to do some feature testing on Adoptium infrastructure.  We want to segregate these pipelines from the release and evaluation pipelines so that they are easy to differentiate and track.

They will have a prefix on the name (much like we do with release- and evaluation- pipelines) and trigger AQAvit testing with a suffix _trsl (suffixes already supported for AQAvit pipelines, _fips, _criu, etc).  In this way, they will also be differentiated and more easily searchable in TRSS.

**Comment by smlambert** [comment]
Will iteratively add documentation [here](https://github.com/adoptium/ci-jenkins-pipelines/wiki/Trestle-Initiative).

**Comment by smlambert** [comment]
Notes:
- Job restriction plugin is in place (related: https://github.com/adoptium/infrastructure/issues/3379)
- https://ci.adoptium.net/job/Trestle_Pipeline/ as placeholder / front end to triggering build/test pipeline

**Comment by sxa** [comment]
I notice that this is running the same platform specific build pipelines underneath the top level ones so it is using the _temurin suffixed ones. Has any consideration been made to using the _hotspot ones instead for trestle? Clearly there are various implications of this (retention disk space being one) but it may be useful to have that separation, particularly since the trestle pipelines have to put in additional build options that would not be required for a "VARIANT=hotspot" build.
I ran a jdk8u pipeline earlier and that fell over due to the Temurin variant builds bedding our extra company name option which doesn't exist on a normal openjdk fork.

**Comment by smlambert** [comment]
Yes, I would prefer to use the  -hotspot variant but hit build failure upon running it, so shifted back in order to build successfully.

**Comment by sxa** [comment]
Hmmm yeah I j[ust did a quick test with one of the hotspot pipelines](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-aix-ppc64-hotspot/lastFailedBuild/) equivalent to [a _temurin one I ran yesterday](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-aix-ppc64-temurin/lastSuccessfulBuild/)) and it fell over with - LMK if this looks similar to what you hit:
```
Loading library openjdk-jenkins-helper@null
Examining [adoptium/jenkins-helper](https://github.com/adoptium/jenkins-helper)
Attempting to resolve null as a branch
Attempting to resolve null as a tag
ERROR: Could not resolve null
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: No version null found for library openjdk-jenkins-helper
/home/jenkins/.jenkins/jobs/build-scripts/jobs/jobs/jobs/jdk8u/workspace/jdk8u-aix-ppc64-hotspot@tmp/jfrog/1202/.jfrog deleted
Finished: FAILURE
```

**Comment by smlambert** [comment]
`ERROR: Could not resolve null` seems familiar, yes

**Comment by sxa** [comment]
> Yes, I would prefer to use the _hotspot variant but hit build failure upon running it, so shifted back to _temurin variant in order to build successfully.

Interesting - would definitely be good to fix that. My initial though is that it might be problematic if the jobs haven't been regenerated in ages so are out of sync...

**Comment by sxa** [comment]
Adding this for reference - there is a proposal in flight just now to avoid doing full testing upstream on mac/x64 which could mean that's a platform that people will be interested in using under Trestle going forward: 

- https://bugs.openjdk.org/browse/JDK-8341097
- https://github.com/openjdk/jdk/pull/21257
